core/src/test/java/com/linecorp/armeria/server/HttpHeaderUtilTest.java

```diff
@@ -22,205 +22,273 @@
-                .isEqualTo(InetAddress.getByName("192.0.2.61"));
+                .containsExactlyElementsOf(ImmutableList.of(new InetSocketAddress("192.0.2.61", 0)));
 
         // The unknown identifier
         assertThat(forwarded("for=unknown;proto=http;by=203.0.113.43,for=192.0.2.61"))
-                .isIn(InetAddress.getAllByName("192.0.2.61"));
+                .containsExactlyElementsOf(ImmutableList.of(new InetSocketAddress("192.0.2.61", 0)));
+    }
+
+    @Nullable
+    private static List<InetSocketAddress> forwarded(String value) {
+        return getAllValidAddresses(value, HttpHeaderUtil.FORWARDED_CONVERTER, ACCEPT_ANY);
+    }
+
+    @Nullable
+    private static List<InetSocketAddress> forwarded(String value, Predicate<InetAddress> clientAddressFilter) {
+        return getAllValidAddresses(value, HttpHeaderUtil.FORWARDED_CONVERTER, clientAddressFilter);
     }
 
     @Nullable
-    private static InetAddress forwarded(String value) {
-        return HttpHeaderUtil.getFirstValidAddress(value, HttpHeaderUtil.FORWARDED_CONVERTER, ACCEPT_ANY);
+    private static List<InetSocketAddress> getAllValidAddresses(
+            String value, Function<String, String> valueConverter, Predicate<InetAddress> clientAddressFilter) {
+        final Builder<InetSocketAddress> builder = ImmutableList.builder();
+        HttpHeaderUtil.getAllValidAddress(value, valueConverter, clientAddressFilter, builder);
+        return builder.build();
     }
 
     @Test
-    public void getAddress_X_Forwarded_For() throws UnknownHostException {
+    void getAddress_X_Forwarded_For() throws UnknownHostException {
         // IPv4
         assertThat(xForwardedFor("192.0.2.60,192.0.2.61,192.0.2.62"))
-                .isEqualTo(InetAddress.getByName("192.0.2.60"));
+                .isEqualTo(ImmutableList.of(new InetSocketAddress("192.0.2.60", 0),
+                                            new InetSocketAddress("192.0.2.61", 0),
+                                            new InetSocketAddress("192.0.2.62", 0)));
 
         // IPv4 with a port number
         assertThat(xForwardedFor("192.0.2.60:4711,192.0.2.61:4711,192.0.2.62:4711"))
-                .isEqualTo(InetAddress.getByName("192.0.2.60"));
+                .isEqualTo(ImmutableList.of(new InetSocketAddress("192.0.2.60", 4711),
+                                            new InetSocketAddress("192.0.2.61", 4711),
+                                            new InetSocketAddress("192.0.2.62", 4711)));
 
         // IPv6
         assertThat(xForwardedFor("[2001:db8:cafe::17],[2001:db8:cafe::18],[2001:db8:cafe::19]"))
-                .isEqualTo(InetAddress.getByName("2001:db8:cafe::17"));
+                .isEqualTo(ImmutableList.of(
+                        new InetSocketAddress("[2001:db8:cafe::17]", 0),
+                        new InetSocketAddress("[2001:db8:cafe::18]", 0),
+                        new InetSocketAddress("[2001:db8:cafe::19]", 0)));
```

Comment:
global comment - Brackets are unnecessary when calling `new InetSocketAddress()`

Reply:
Thanks for education.